### ISSUE TITLE
Removing Rancher/Cleaning up Clusters (in Rancher and which has Rancher installed)

### ISSUE THREAD

**Issue by deniseschannon** [rancher/docs#779]
Make a new section that covers how to remove Rancher from a k8s cluster.
ANd then move cleaning cluster nodes under this new section.

**Comment by dramich** [comment]
Starting in 2.1 the label `cattle.io/creator=norman` will be added to every resource in a cluster that is created by norman. Prior to 2.1 there is not a deterministic way to tell if we created anything or simply updated it to add our info.

The following only applies to deleting an imported clusters. If a user deletes a different type of cluster we do different things. Managed clusters are deleted from the provider, IaaS deletes the host etc.

---
From the UI/API:

When a user deletes an imported cluster, rancher server is going to create 4 new resources in the user cluster:
1. serviceAccount
2. clusterRole
3. clusterRoleBinding
4. job

The first three resources are all created with the ownerReference being set to the job. This is done so k8s will delete the job's dependencies when the job is deleted. 

The job runs the agent image in a cleanup mode. After sleeping for 1 min to allow the rancher server to finish cleanup and disconnect the job will start performing the below tasks:

All versions of rancher server:
Remove the cattle-system namespace. This kills the agent and stops all communication with the rancher server.
Cleanup all other namespaces - remove finalizers, annotations and labels except the label `cattle.io/creator=norman`.

2.1+
Remove roles, roleBindings, clusterRoles and clusterRoleBindings that rancher created.

Assuming the job completes successfully it will delete itself and k8s will cleanup it's dependencies. If it does not complete it will restart and run until it hits the backoffLimit (by default this is 6) see https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#pod-backoff-failure-policy. In this case the job will not be deleted and the user can either check the pod logs for info on why or simply delete the job. 

---

Using the script:
The script lives at `rancher/rancher/cleanup/user-cluster.sh`

User would need to grab the script and make executable.
Run the script using `./user-cluster.sh {AGENT_IMAGE} {FLAGS}`
The AGENT_IMAGE is required. If the docs can default this to the latest version that would be good. Current version would be rancher/rancher-agent:v2.1.0 after it's released.
The flag `-dry-run` will run through all resources and print any changes it would potentially make without actually modifying anything. 

If this is for an airgap install the user would need to update the script by replacing the `yaml_url` with wherever they are able to put the `user-cluster.yml` file. Our version is located in `rancher/rancher/cleanup/user-cluster.yml`

The script runs the same job as above, just a different way to call it.
